src/js/components/AppListFilterComponent.jsx

```diff
@@ -22,9 +23,13 @@ var AppListFilterComponent = React.createClass({
       filterText: filterText,
       activated: filterText !== "" || this.state.focused
     });
-    this.props.onChange(filterText);
+    this.fireChangeEvent(filterText);
```

Comment:
How about this:

```
fireChangeEvent: _.debounce(() => {
  this.props.onChange(this.state.filterText);
}, 100),

// ...

this.setState({
  filterText: filterText,
  activated: filterText !== "" || this.state.focused
}, this.fireChangeEvent);
```
The reason for this suggestion is that it ensures consistency: the value being sent to `this.props.onChange` is guaranteed to be the one currently in `this.state.filterText`. 

From [the React docs](https://facebook.github.io/react/docs/component-api.html#setstate):
> There is no guarantee of synchronous operation of calls to setState and calls may be batched for performance gains.